### PR TITLE
py sol for DMOJ Spirale

### PR DESCRIPTION
**Pull request by freakin23**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

AC: https://dmoj.ca/submission/6717894

### CODE REVIEW

File: solutions/bronze/dmoj-spirale.mdx [hunk 1]
```diff
@@ -73,3 +73,64 @@ int main() {
...
+</CPPSection>
+<PySection>
+
+```py
+ux = [0, -1, 0, 1]
+uy = [1, 0, -1, 0]
+dx = [0, -1, 0, 1]
+dy = [-1, 0, 1, 0]
+
+mat = [[(float("inf"))] * 50 for _ in range(50)]
```
Comment (SansPapyrus683) [inline solutions/bronze/dmoj-spirale.mdx:86]: magic #

Comment (freakin23) [reply solutions/bronze/dmoj-spirale.mdx]: added

Comment (freakin23) [reply solutions/bronze/dmoj-spirale.mdx]: AC: https://dmoj.ca/submission/6718196

File: solutions/bronze/dmoj-spirale.mdx [hunk 2]
```diff
@@ -73,3 +73,65 @@ int main() {
...
+```py
+ux = [0, -1, 0, 1]
+uy = [1, 0, -1, 0]
+dx = [0, -1, 0, 1]
+dy = [-1, 0, 1, 0]
+
+MAX_N = 50
+mat = [[(float("inf"))] * MAX_N for _ in range(MAX_N)]
+curr = 0
+cnt = 0
```
Comment (SansPapyrus683) [inline solutions/bronze/dmoj-spirale.mdx:89]: better name

Comment (freakin23) [reply solutions/bronze/dmoj-spirale.mdx:89]: for `curr` and `cnt`?

File: solutions/bronze/dmoj-spirale.mdx (hunk 1 above)
Comment (TheGamingMousse) [reply solutions/bronze/dmoj-spirale.mdx]: > AC: https://dmoj.ca/submission/6718196

fyi dmoj submissions are private unless you solved the problem

File: solutions/bronze/dmoj-spirale.mdx [hunk 3]
```diff
@@ -10,66 +10,132 @@ author: Oscar Garries
...
+
+const int MAX_N = 50;
+int mat[MAX_N][MAX_N], visited_cells_cnt = 0, n, m, k, current_distance = 0;
 
-void fn(int x, int y) {
+void update_distance(int x, int y) {
 	if (x < 0 || x >= m || y < 0 || y >= n) return;
-	cnt++;
-	mat[y][x] = min(mat[y][x], cur);
+	visited_cells_cnt++;
```
Comment (SansPapyrus683) [inline solutions/bronze/dmoj-spirale.mdx:28]: just `num_visited` is fine

Comment (freakin23) [reply solutions/bronze/dmoj-spirale.mdx]: done

**Review by SansPapyrus683 — APPROVED** [review]
well the c++ code is just as bad

**Comment by freakin23** [comment]
> well the c++ code is just as bad

maybe it is bcz of lots of variables are used.

**Pull request merged**